autotest/pst_from_tests.py

```diff
@@ -4886,274 +4886,479 @@ def list_float_int_index_test(tmp_path):
+                                      pargp=[prefix+"threshproportion_k:{0}".format(k),prefix+"threshfill_k:{0}".format(k)],
+                                      lower_bound=[0.1,0.1],upper_bound=[10.0,10.0],transform="none",par_style='d')
+
+                    pf.add_observations(arr_file,prefix="hkarr-"+prefix+"_k:{0}".format(k),
+                                        obsgp="hkarr-"+prefix+"_k:{0}".format(k),zone_array=ib)
+
+                    pf.add_observations(arr_file+".threshcat.dat", prefix="tcatarr-" + prefix+"_k:{0}".format(k),
+                                        obsgp="tcatarr-" + prefix+"_k:{0}".format(k),zone_array=ib)
+
+                    pf.add_observations(arr_file + ".thresharr.dat",
+                                        prefix="tarr-" +prefix+"_k:{0}".format(k),
+                                        obsgp="tarr-" + prefix + "_k:{0}".format(k), zone_array=ib)
+
+                    df = pd.read_csv(threshcsv.replace(".csv","_results.csv"),index_col=0)
+                    pf.add_observations(os.path.split(threshcsv)[1].replace(".csv","_results.csv"),index_cols="threshcat",use_cols=df.columns.tolist(),prefix=prefix+"-results_k:{0}".format(k),
+                                        obsgp=prefix+"-results_k:{0}".format(k),ofile_sep=",")
+                    num_cat_arrays += 1
 
-    pst.control_data.noptmax = 0
-    pst.pestpp_options["additional_ins_delimiters"] = ","
+        # add model run command
+        pf.mod_sys_cmds.append("mf6")
+        print(pf.mult_files)
+        print(pf.org_files)
 
-    pst.write(os.path.join(pf.new_d, "freyberg.pst"))
-    pyemu.os_utils.run("{0} freyberg.pst".format(ies_exe_path), cwd=pf.new_d)
+        # build pest
+        pst = pf.build_pst('freyberg.pst')
+        #cov = pf.build_prior(fmt="none")
+        #cov.to_coo(os.path.join(template_ws, "prior.jcb"))
+        pst.try_parse_name_metadata()
 
-    res_file = os.path.join(pf.new_d, "freyberg.base.rei")
-    assert os.path.exists(res_file), res_file
-    pst.set_res(res_file)
-    print(pst.phi)
-    assert pst.phi < 0.1, pst.phi
+        pst.control_data.noptmax = 0
+        pst.pestpp_options["additional_ins_delimiters"] = ","
 
-    par = pst.parameter_data
-    cat1par = par.loc[par.threshcat=="0","parnme"]
-    cat2par = par.loc[par.threshcat=="1","parnme"]
-    print(cat1par,cat2par)
-    assert cat1par.shape[0] == 3
-    assert cat2par.shape[0] == 3
+        pst.write(os.path.join(pf.new_d, "freyberg.pst"))
+        pyemu.os_utils.run("{0} freyberg.pst".format(ies_exe_path), cwd=pf.new_d)
 
-    par.loc[cat1par,"parval1"] = 0.1
-    par.loc[cat1par, "parubnd"] = 1.0
-    par.loc[cat1par, "parlbnd"] = 0.01
+        res_file = os.path.join(pf.new_d, "freyberg.base.rei")
+        assert os.path.exists(res_file), res_file
+        pst.set_res(res_file)
+        print(pst.phi)
+        assert pst.phi < 0.1, pst.phi
```

Comment:
if this is expected to be "near zero" we should be able to be more strict. #498 should help with this.